rpadmin/admin.go

```diff
@@ -630,3 +632,48 @@ func MaxRetries(r int) Opt {
 		cl.MaxRetries = r
 	}}
 }
+
+// AdminAddressesFromK8SDNS attempts to deduce admin API URLs
+// based on Kubernetes DNS resolution.
+// We are in Kubernetes.
+// https://github.com/kubernetes/dns/blob/master/docs/specification.md
+// Assume that Admin API URL configured is a Kubernetes Service URL.
+// This Admin API URL is passed in as the function argument.
+// Since it's a Kubernetes service, Kubernetes DNS creates a DNS SRV record
+// for the admin port mapping.
+// We can query the DNS record to get the target host names and ports.
+func AdminAddressesFromK8SDNS(adminAPIURL string) ([]string, error) {
+	adminURL, err := url.Parse(adminAPIURL)
+	if err != nil {
+		return nil, err
+	}
+
+	_, records, err := net.LookupSRV("admin", "tcp", adminURL.Hostname())
+	if err != nil {
+		return nil, err
+	}
+
+	if len(records) == 0 {
+		return nil, nil
+	}
```

Comment:
I'm afraid we can run into nil pointer exceptions if not carefully checking the response. Should we return a checkable error in the case where we couldn't get any results. Do we end up here if the SRV record didn't exist as well or only if we got a response that doesn't map to any targets?

Reply:
I added `ErrNoSRVRecordsFound` error that is returned in case of no records... I am not really able to test this and mock it. I think this would be perhaps some weird edge case if possible at all... My intuition is that is the SRV records do not exist, we should get a [DNS error](https://pkg.go.dev/net#DNSError) since the query should return `NXDOMAIN`.